plugins/guests/redhat/cap/osvariant.rb

```diff
@@ -3,13 +3,14 @@ module GuestRedHat
     module Cap
       class OsVariant
         def self.os_variant(machine)
-          machine.communicate.sudo("grep VARIANT_ID /etc/os-release") do |type, data|
-            if type == :stderr
-              @env.ui.error(data)
-              exit 126
+          command = "grep VARIANT_ID /etc/os-release"
+          # TODO: execute efficient command to solve this
+          if machine.communicate.test(command) # test if command is exits with code 0
+            machine.communicate.execute(command) do |type, data|
+              return data.chomp.gsub(/"/, '').split("=").last
             end
-            return data.chomp.gsub(/"/, '').split("=").last
           end
+          return nil
```

Comment:
Probably this line is not required. 
If `if` block doesn't execute then `nil` will be returned by default.